tests/externalsystems/OfferProvider.Library.Tests/BusinessLogic/OfferProviderBusinessLogicTests.cs

```diff
@@ -296,7 +322,7 @@ public async Task TriggerProviderCallback_WithValidData_ExternalSA_ReturnsExpect
             .MustNotHaveHappened();
         A.CallTo(() => _provisioningManager.GetCentralClientAuthDataAsync(A<string>._))
             .MustNotHaveHappened();
-        A.CallTo(() => _offerProviderService.TriggerOfferProviderCallback(A<OfferProviderCallbackData>.That.Matches(x => x.TechnicalUserInfo!.TechnicalUserSecret == null), A<string>._, A<CancellationToken>._))
+        A.CallTo(() => _offerProviderService.TriggerOfferProviderCallback(A<OfferProviderCallbackData>.That.Matches(x => x.TechnicalUsersInfo!.FirstOrDefault()!.TechnicalUserSecret == null), A<string>._, A<CancellationToken>._))
```

Comment:
inappropriate use of !
```suggestion
        A.CallTo(() => _offerProviderService.TriggerOfferProviderCallback(A<OfferProviderCallbackData>.That.Matches(x => x.TechnicalUsersInfo != null && x.TechnicalUsersInfo.Count() > 0 && x.TechnicalUsersInfo.First().TechnicalUserSecret == null), A<string>._, A<CancellationToken>._))
```